sklearn/preprocessing/label.py

```diff
@@ -473,9 +474,19 @@ def label_binarize(y, classes, neg_label=0, pos_label=1, sparse_output=False):
 
     if y_type == "binary":
         if len(classes) == 1:
-            Y = np.zeros((len(y), 1), dtype=np.int)
-            Y += neg_label
-            return Y
+            if sparse_output:
+                # To handle [0, 0, ...] or ['0', '0', ...] case
+                if (isinstance(classes[0], (numbers.Integral, np.integer))
+                        and classes[0] == 0) or classes[0] == '0':
+                    classes = np.r_[pos_label, classes]
+                    n_classes += 1
+                else:
+                    classes = np.r_[neg_label, classes]
+                    n_classes += 1
```

Comment:
Can you again clarify why you are branching out here?

If `len(classes=1)` it should always be encoded with the positive label.